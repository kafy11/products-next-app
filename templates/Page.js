import PageHead from './PageHead';
import PageFooter from './PageFooter';

const Page = ({ title, children, toolbarContent }) => (
  <>
    <PageHead title={title}/>
    <div className="container py-5">
      <div className="row">
        <div className="col-auto">
          <h3>{title}</h3>
        </div>
        <div className="col">
          {toolbarContent}
        </div>
      </div>
      
      {children}

      <PageFooter />
    </div>
  </>
);

export default Page;