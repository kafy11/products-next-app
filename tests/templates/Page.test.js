import { shallow } from 'enzyme';
import Page from '../../templates/Page';

const props = {
    title: 'Test',
    children: 'Children test',
    toolbarContent: 'Toolbar test'
};

let wrapper;
beforeEach(() => {
    wrapper = shallow(<Page {...props}/>);
});

describe('Page', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should render the title correctly', () => {
        expect(wrapper.find('.page__title').text()).toBe(props.title);
    });
    
    it('should render the content correctly', () => {
        expect(wrapper.find('.page__content').text()).toBe(props.children);
    });
    
    it('should render the toolbar content correctly', () => {
        expect(wrapper.find('.page__toolbar').text()).toBe(props.toolbarContent);
    });
});