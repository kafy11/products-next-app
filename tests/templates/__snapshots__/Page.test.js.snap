// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render correctly 1`] = `
<Fragment>
  <PageHead
    title="Test"
  />
  <div
    className="container py-5"
  >
    <div
      className="row"
    >
      <div
        className="col-auto"
      >
        <h3
          className="page__title"
        >
          Test
        </h3>
      </div>
      <div
        className="col page__toolbar"
      >
        Toolbar test
      </div>
    </div>
    <div
      className="page__content"
    >
      Children test
    </div>
    <PageFooter />
  </div>
</Fragment>
`;
